inc/class-wpseo-image-utils.php

```diff
@@ -26,8 +26,42 @@ public static function get_attachment_by_url( $url ) {
 			// @codeCoverageIgnoreEnd -- The rest we _can_ test.
 		}
 
-		// phpcs:ignore WordPress.VIP.RestrictedFunctions -- We use the WP COM version if we can, see above.
-		return (int) attachment_url_to_postid( $url );
+		return self::attachment_url_to_postid( $url );
+	}
+
+	/**
+	 * Implements the attachment_url_to_postid with use of WP Cache.
+	 *
+	 * @param string $url The attachment URL for which we want to know the Post ID.
+	 *
+	 * @return int The Post ID belonging to the attachment, 0 if not found.
+	 */
+	protected static function attachment_url_to_postid( $url ) {
+		$cache_key = sprintf( 'yoast_attachment_url_post_id_%s', md5( $url ) );
+
+		// Set the id based on the hashed url in the cache.
+		$id = wp_cache_get( $cache_key );
+
+		if ( $id === 'not_found' ) {
+			return 0;
+		}
+
+		// ID is found in cache, return.
+		if ( $id !== false ) {
+			return $id;
+		}
+
+		$id = attachment_url_to_postid( $url );
```

Comment:
Can you add this comment above this line: 
`// phpcs:ignore WordPress.VIP.RestrictedFunctions -- We use the WP COM version if we can, see above.`